File: server/src/main/java/de/ketrwu/portfolio/forms/ContactForm.java
package de.ketrwu.portfolio.forms;


import lombok.Data;

import javax.validation.constraints.AssertTrue;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.Size;

@Data
public class ContactForm extends CaptchaForm {

    @NotEmpty(message = "E-Mail Adresse ist ein Pflichtfeld.")
    @Email(message = "E-Mail Adresse ist nicht korrekt.")
    private String email;

    @NotEmpty(message = "Nachricht ist ein Pflichtfeld")
    @Size(min = 10, message = "Nachricht ist zu kurz. Mindestens 10 Zeichen.")
    private String message;

    @AssertTrue(message = "Dem Datenschutzhinweis muss zugestimmt werden.")
    private boolean privacyAgreement = false;

}

File: server/src/main/kotlin/de/ketrwu/portfolio/forms/ContactForm.kt
package de.ketrwu.portfolio.forms

import javax.validation.constraints.AssertTrue
import javax.validation.constraints.Email
import javax.validation.constraints.NotEmpty
import javax.validation.constraints.Size

/**
 * @author Kenneth Wußmann
 */
class ContactForm : CaptchaForm() {

    @NotEmpty(message = "E-Mail Adresse ist ein Pflichtfeld.")
    @Email(message = "E-Mail Adresse ist nicht korrekt.")
    var email: String? = null

    @NotEmpty(message = "Nachricht ist ein Pflichtfeld")
    @Size(min = 10, message = "Nachricht ist zu kurz. Mindestens 10 Zeichen.")
    var message: String? = null

    @AssertTrue(message = "Dem Datenschutzhinweis muss zugestimmt werden.")
    var privacyAgreement = false
}